processing/src/main/java/org/apache/druid/query/filter/Filter.java

```diff
@@ -129,4 +131,52 @@ default boolean canVectorizeMatcher()
   {
     return false;
   }
+
+  /**
+   * Set of columns used by a filter
+   */
+  Set<String> getRequiredColumns();
+
+  /**
+   * "Manual" {@link FilterTuning} to allow explicit control of filter optimization behavior. This will likely be
+   * supplied by the {@link DimFilter} that is creating this {@link Filter}
```

Comment:
Is it intended that anything outside of `computeTuning` call this method? If not, please update the javadocs to tell people not to call it. Maybe mention that you would have made it `protected`, except that `Filter` is an interface, so you can't do that.

Reply:
this method has been removed